src/components/schedule/ScheduleList.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { DataGrid, GridRowsProp, GridColDef, GridToolbar } from '@mui/x-data-grid';
+import { courseScheduleData } from '../common/sampleData/courseSchedule'
+import WeekdayTable  from './WeekdayTable'
+import { Course } from '../common/types'
+
+const rows: GridRowsProp = courseScheduleData.map((course: Course, index: number) => ({
+  id: index + 1,
+  term: course.Term,
+  course: course.Subj + ' ' + course.Num,
+  section: course.Section,
+  title: course.Title,
+  scheduleType: course.SchedType,
+  instructor: course.Instructor,
+  location: course.Bldg + ' ' + course.Room,
+  start: convertToTime(course.Begin),
+  end: convertToTime(course.End),
+  days: course.Days,
+  capacity: course.Cap,
+}));
+
+const columns: GridColDef[] = [
+  { field: 'term', headerName: 'Term', width: 100 },
+  { field: 'course', headerName: 'Course', width: 150 },
+  { field: 'section', headerName: 'Section', width: 100 },
+  { field: 'instructor', headerName: 'Instructor', width: 150 },
+  { field: 'capacity', headerName: 'Capacity', type: 'number', width: 100 },
+  { field: 'location', headerName: 'Location', width: 150 },
+  {
+    field: 'days',
+    headerName: 'Days',
+    width: 200,
+    renderCell: (params) => <WeekdayTable days = {params.value.split('')} />,
+  },
+  { field: 'start', headerName: 'Start', width: 100 },
+  { field: 'end', headerName: 'End', width: 100 }
+];
+
+const ScheduleList = () => {
+    return (
+        <div style={{ height: '100vh', width: '100%' }}>
+            <DataGrid 
+              rows={rows} 
+              columns={columns}
+              initialState={{
+                sorting: {
+                  sortModel: [{ field: 'course', sort: 'asc' }],
+                },
+              }}
+              slots={{ toolbar: GridToolbar }}
+            />
+        </div>
+    )
+}
+
+export default ScheduleList
+
+
+// convert time from hhmm to hh:mm
+function convertToTime(minutes: number): string {
```

Comment:
I would use a library to do this, expecially considering we will likely need to perform date/time operations in other components later on. I believe `date-fns` is the most popular solution these days: https://www.npmjs.com/package/date-fns

Reply:
Change the function to use `date-fns` library in the latest commit